merlin/models/tf/metrics/topk.py

```diff
@@ -282,43 +297,126 @@ def from_config(cls, config):
         fn = config.pop("fn", None)
         k = config.pop("k", None)
         pre_sorted = config.pop("pre_sorted", None)
-        if cls is RankingMetric:
+        if cls is TopkMetric:
             return cls(get_metric(fn), k=k, pre_sorted=pre_sorted, **config)
-        return super(RankingMetric, cls).from_config(config)
+        return super(TopkMetric, cls).from_config(config)
 
 
 @metrics_registry.register_with_multiple_names("recall_at", "recall")
-class RecallAt(RankingMetric):
+class RecallAt(TopkMetric):
     def __init__(self, k=10, pre_sorted=False, name="recall_at"):
         super().__init__(recall_at, k=k, pre_sorted=pre_sorted, name=name)
 
 
 @metrics_registry.register_with_multiple_names("precision_at", "precision")
-class PrecisionAt(RankingMetric):
+class PrecisionAt(TopkMetric):
     def __init__(self, k=10, pre_sorted=False, name="precision_at"):
         super().__init__(precision_at, k=k, pre_sorted=pre_sorted, name=name)
 
 
 @metrics_registry.register_with_multiple_names("map_at", "map")
-class AvgPrecisionAt(RankingMetric):
+class AvgPrecisionAt(TopkMetric):
     def __init__(self, k=10, pre_sorted=False, name="map_at"):
         super().__init__(average_precision_at, k=k, pre_sorted=pre_sorted, name=name)
 
 
 @metrics_registry.register_with_multiple_names("mrr_at", "mrr")
-class MRRAt(RankingMetric):
+class MRRAt(TopkMetric):
     def __init__(self, k=10, pre_sorted=False, name="mrr_at"):
         super().__init__(mrr_at, k=k, pre_sorted=pre_sorted, name=name)
 
 
 @metrics_registry.register_with_multiple_names("ndcg_at", "ndcg")
-class NDCGAt(RankingMetric):
+class NDCGAt(TopkMetric):
     def __init__(self, k=10, pre_sorted=False, name="ndcg_at"):
         super().__init__(ndcg_at, k=k, pre_sorted=pre_sorted, name=name)
 
 
-def ranking_metrics(top_ks: Sequence[int], **kwargs) -> Sequence[RankingMetric]:
-    metrics: List[RankingMetric] = []
+class TopKMetricsAggregator(Metric, TopkMetricWithLabelRelevantCountsMixin):
+    """Aggregator for top-k metrics (TopkMetric) that is optimized
+    to sort top-k predictions only once for all metrics.
+
+    topk_metrics : Sequence[TopkMetric]
+        Multiple arguments with TopkMetric instances
+    """
+
+    def __init__(self, topk_metrics: Sequence[TopkMetric]):
```

Comment:
```suggestion
    def __init__(self, *topk_metrics: TopkMetric):
```

Reply:
This would simplify the API a bit. Since you could do: `TopKMetricsAggregator(RecallAt(5), MRRAt(5))`, instead of having to wrap it into a list.

Reply:
Done